ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -249,8 +253,33 @@ public void setEventListener(ReactRootViewEventListener eventListener) {
     return Assertions.assertNotNull(mJSModuleName);
   }
 
-  /* package */ @Nullable Bundle getLaunchOptions() {
-    return mLaunchOptions;
+  public @Nullable Bundle getAppProperties() {
+    return mAppProperties;
+  }
+
+  public void setAppProperties(@Nullable Bundle appProperties) {
+    mAppProperties = appProperties;
+
+    if (mReactInstanceManager == null || !mIsAttachedToInstance ||
+      mReactInstanceManager.getCurrentReactContext() == null) {
+        return;
+    }
+
+    this.runApplication(mReactInstanceManager);
+  }
+
+  /* package */ void runApplication(final ReactInstanceManager reactInstanceManager) {
```

Comment:
Why do you need to pass in reactInstanceManager if it's already available as `mReactInstanceManager`?